gateway/src/main/kotlin/DefaultGateway.kt

```diff
@@ -61,21 +48,16 @@ public data class DefaultGatewayData(
 /**
  * The default Gateway implementation of Kord, using an [HttpClient] for the underlying webSocket
  */
-public class DefaultGateway(private val data: DefaultGatewayData) : Gateway {
-
-    override val coroutineContext: CoroutineContext = SupervisorJob() + data.dispatcher
+public class DefaultGateway(private val data: DefaultGatewayData) : BaseGateway() {
 
     private val compression: Boolean
-
     private val _ping = MutableStateFlow<Duration?>(null)
-    override val ping: StateFlow<Duration?> get() = _ping
 
-    override val events: SharedFlow<Event> = data.eventFlow
+    override val ping: StateFlow<Duration?> get() = _ping
+    override val events: MutableSharedFlow<Event> = data.eventFlow
```

Comment:
We should be aware that this now exposes a MutableSharedFlow. which means it can be interred with.
While I'm not against it since it would mean the user of Kord may emit their own custom Events through.

Unless I'm missing something.

Reply:
I guess if someone would relay on implementation rather than the interface they should be aware of what they are doing and the flexibility should not harm?

Reply:
I didn't say it's a bad change I'm just pointing out the change :D